app/views/events/_form.html.erb

```diff
@@ -1,97 +1,91 @@
-      <div class="col-lg-10">
-        <%= f.select(:game_mode, League.game_modes.map {|k, v| [k.humanize, v]},
-                     {:include_blank => t('events.new.select_game_mode')},
-                     { :class => 'form-control', :id => 'event_game_mode_league'}) %>
-      </div>
-    </div>
-    <div class="form-group" id="tournament-game-mode">
-      <%= f.label Event.human_attribute_name(:game_mode), :class => 'control-label col-lg-2' %>
-      <div class="col-lg-10">
-        <%= f.select(:game_mode, Tournament.game_modes.map {|k, v| [k.humanize, v]},
-                     {:include_blank => t('events.new.select_game_mode')},
-                     { :class => 'form-control', :id => 'event_game_mode_tournament'}) %>
-      </div>
+      <% case @event.type %>
+      <% when 'League' %>
+          <%= f.label :game_mode, class: 'control-label col-lg-2' %>
+          <div class="col-lg-10">
+            <%= f.select(:game_mode, League.game_modes.map {|k, v| [k.humanize, v]},
+                         { include_blank: t('events.new.select_game_mode')},
+                         { class: 'form-control'}) %>
+          </div>
+      <% when 'Tournament' %>
+          <%= f.label :game_mode, class: 'control-label col-lg-2' %>
+          <div class="col-lg-10">
+            <%= f.select(:game_mode, Tournament.game_modes.map {|k, v| [k.humanize, v]},
+                         { include_blank: t('events.new.select_game_mode') },
+                         { class: 'form-control' }) %>
+          </div>
+      <% end %>
     </div>
     <div class="form-group">
-      <%= f.label Event.human_attribute_name(:player_type), :class => 'control-label col-lg-2' %>
+      <%= f.label :player_type, class: 'control-label col-lg-2' %>
       <div class="col-lg-10">
         <%= f.select(:player_type, Event.player_types.map {|k, v| [k.humanize, v]},
-                     {:include_blank => t('events.new.select_player_type')},
-                     { :class => 'form-control'}) %>
+                     {include_blank: t('events.new.select_player_type')},
+                     { class: 'form-control'}) %>
       </div>
     </div>
     <div class="form-group">
-      <%= f.label Event.human_attribute_name(:max_teams), :class => 'control-label col-lg-2' %>
+      <%= f.label :max_teams, class: 'control-label col-lg-2' %>
       <div class="col-lg-10">
-        <%= f.text_field :max_teams, id: :event_max_teams, :class => 'form-control' %>
+        <%= f.text_field :max_teams, class: 'form-control' %>
         <%= f.error_span(:max_teams) %>
       </div>
     </div>
     <div class="form-group">
-      <%= f.label Event.human_attribute_name(:discipline), :class => 'control-label col-lg-2' %>
+      <%= f.label :discipline, class: 'control-label col-lg-2' %>
       <div class="col-lg-10">
-        <%= f.text_field :discipline, id: :event_discipline, :class => 'form-control' %>
```

Comment:
why are the id's not needed anymore?

Reply:
Because the form helpers automatically set correct ids. We also used that in the introduction to rails.